[BUG] go build fails in version 3.4.7

### Is there an existing issue for this?

- [x] I have searched the existing issues.

### Current Behavior

Building nuclei with these params, fails.

Build command:
```
go build -o nuclei --tags=stats
```

### Expected Behavior

It works in previous verions like v3.3.7

### Steps To Reproduce

Simply build like this

```
go build -o nuclei --tags=stats
```

### Relevant log output

```shell
107.5 # github.com/projectdiscovery/nuclei/v3/pkg/scan/events
107.5 ../../pkg/scan/events/stats_build.go:62:10: cannot use json.NewEncoder(f) (value of interface type sonic.Encoder) as *"github.com/projectdiscovery/nuclei/v3/pkg/utils/json".Encoder value in assignment: sonic.Encoder does not implement *"github.com/projectdiscovery/nuclei/v3/pkg/utils/json".Encoder (type *"github.com/projectdiscovery/nuclei/v3/pkg/utils/json".Encoder is pointer to interface, not interface)
107.5 ../../pkg/scan/events/stats_build.go:71:15: s.enc.Encode undefined (type *"github.com/projectdiscovery/nuclei/v3/pkg/utils/json".Encoder is pointer to interface, not interface)
```

### Environment

```markdown
- OS: Linux
- Nuclei: 3.4.7
- Go: 1.24
```

### Anything else?

_No response_

--- a/pkg/scan/events/stats_build.go
+++ b/pkg/scan/events/stats_build.go
@@ -25,7 +25,7 @@ type ScanStatsWorker struct {
 	m         *sync.Mutex
 	directory string
 	file      *os.File
-	enc       *json.Encoder
+	enc       json.Encoder
 }
 
 // Init initializes the scan stats worker

